playground/frontend/integration_test/standalone_cancel_running_example_test.dart

```diff
@@ -49,15 +43,8 @@ void main() {
 
 Future<void> _runAndCancelExample(WidgetTester wt, Duration duration) async {
   await wt.tap(find.runOrCancelButton());
-  try {
-    await wt.pumpAndSettle(
-      const Duration(milliseconds: 100),
-      EnginePhase.sendSemanticsUpdate,
-      duration,
-    );
-  } catch (e) {
-    //ignore
-  }
+
+  await wt.wait(duration);
```

Comment:
Investigate.

Reply:
done

Reply:
This worked for me instead, and the test is green:

```dart
  await wt.pump();
  await Future.delayed(duration);
```

Can this be used instead of this function? If not, then why?

Reply:
this approach freezes main isolate (watch the animation on the button with both approaches)

Reply:
To me this is a lesser evil than inventing a method with a vague name without being sure how exactly it does what it does. We can just test the result.

Reply:
What is the definition of "freeze"?

I believe it is impossible to put the thread to sleep in JavaScript.
I guess what appears "frozen" is just animations ticking but not ticking or ticking but not rendering.

If `pump()` ticks animations once and schedules a render, then for the test purpose there technically is no difference between calling it once vs calling it in a loop, it's just that with the loop we are visually satisfied with smooth animation at the cost of active waiting and complexity. I suggest keeping things simple.

Reply:
added `pumpAndSettleNoException`